gossip/ethapi_backend.go

```diff
@@ -355,18 +355,22 @@ func (b *EthAPIBackend) GetReceiptsByNumber(ctx context.Context, number rpc.Bloc
 	number = rpc.BlockNumber(blockNumber)
 
 	block := b.state.GetBlock(common.Hash{}, uint64(number))
+	return b.MakeReceiptsFromBlock(block), nil
+}
+
+func (b *EthAPIBackend) MakeReceiptsFromBlock(block *evmcore.EvmBlock) types.Receipts {
```

Comment:
Why "MakeReceiptsFromBlock" instead of "GetReceiptsForBlock"?

Make kinda implies that all the data is in the block, and the function is just extracting it from the block and re-formatting it. Get, on the other hand, implies that there are data base accesses involved.